class-wp-bootstrap-navwalker.php

```diff
@@ -245,7 +245,7 @@ public function start_el( &$output, $item, $depth = 0, $args = array(), $id = 0
 			}
 
 			/** This filter is documented in wp-includes/post-template.php */
-			$title = apply_filters( 'the_title', esc_html( $item->title ), $item->ID );
+			$title = apply_filters( 'the_title', $item->title, $item->ID );
```

Comment:
Since this is removing an escape could you add a little comment in the code noting that this is capable of outputting html?

Also do you think an alternative sanitation should be run to prevent anything malicious making it out to the page?

Reply:
I thought about how one could do sanitation and had no idea. I then decided that doing no escaping at all is fine as there is no escaping in the `Walker_Nav_Menu` either (it even allows to put javascript in the title which gets executed...). So if something malicious happens here, it most likely also happens on the page with the same title.
The native walker leaves it to the theme developer who can use the `the_title` and the `nav_menu_item_title` filter hooks to do proper escaping in accordance with the theme.

I can add a comment tomorrow, but this means just telling people that this walker behaves like the default walker.

Reply:
What about doing this?

```diff
...
$title = apply_filters( 'nav_menu_item_title', $title, $item, $args, $depth );

+ $title = self::esc_nav_menu_item_title( $title );
...
```
with
```php
/**
 * Escapes the nav menu item title.
 *
 * @see https://developer.wordpress.org/reference/functions/wp_kses_allowed_html/
 *
 * @param string $nav_menu_item_title The menu item's title.
 * @return string
 */
private function esc_nav_menu_item_title( $nav_menu_item_title ) {
	/**
	 * Filters whether to escape the nav menu item title.
	 *
	 * @param bool $escape Whether to escape the nav menu item title.
	 */
	$escape_title = apply_filters( 'wp_bootstrap_navwalker_escape_nav_menu_item_title', $escape = true );
	if ( empty( $nav_menu_item_title ) || ! $escape_title ) {
		return $nav_menu_item_title;
	}

	// Restrict allowed html tags to tags which are considered safe for posts.
	$allowed_tags = wp_kses_allowed_html( 'post' );

	/**
	 * Filters the HTML that is allowed for the nav menu item title.
	 *
	 * @param array $allowed_tags Array of default allowable HTML tags.
	 */
	$allowed_tags = apply_filters( 'wp_bootstrap_allowed_nav_menu_item_title_tags', $allowed_tags );

	return wp_kses( $nav_menu_item_title, $allowed_tags );
}
```
It preserves those HTML tags which are allowed in posts (even if they do not make sense in a menu context) but it prevents scripts from being executed. It can be turned off completely via the `wp_bootstrap_navwalker_escape_nav_menu_item_title` hook and allowed tags can be altered via the `wp_bootstrap_allowed_nav_menu_item_title_tags` hook.

Reply:
After reconsideration I am in agreement with you that following what the default walker does is the best option. No need to add any adjustments here. Will merge as-is and can revisit in future to refactor if necessary :)